inffast.c

```diff
@@ -127,36 +122,25 @@ void Z_INTERNAL zng_inflate_fast(PREFIX3(stream) *strm, unsigned long start) {
                                 /*  window position, window bytes to copy */
     unsigned len;               /* match length, unused bytes */
     unsigned dist;              /* match distance */
-    unsigned char *from;        /* where to copy match from */
-    unsigned extra_safe;        /* copy chunks safely in all cases */
 
     /* copy state to local variables */
     state = (struct inflate_state *)strm->state;
     in = strm->next_in;
     last = in + (strm->avail_in - (INFLATE_FAST_MIN_HAVE - 1));
-    out = strm->next_out;
-    beg = out - (start - strm->avail_out);
-    end = out + (strm->avail_out - (INFLATE_FAST_MIN_LEFT - 1));
-    safe = out + strm->avail_out;
+    wsize = state->wsize;
+    out = state->window + wsize + state->wnext;
+    end = state->window + (wsize * 2) - STD_MAX_MATCH;
+    safe = end - state->chunksize;
```

Comment:
This doesn't look right...

Reply:
`out` makes sense to me, it is writing to the second part of the `state->window` memory allocation. So at `wsize` (window size) + `state->wnext` which is number of bytes written out.

One thing I have thought of was using a `s->pending` similar to what deflate uses, so the calculation `state->window + wsize` doesn't have to keep being done, however it changes more lines of code, so I thought I would leave that change for another PR if we wanted to do something like that.

Reply:
Recalculating first part of `out` (`state->window + wsize`) every time is kinda pointless. The original code didn't do that.

I'm not sure what's the point of subtracting `state->chunksize` from `safe` as it is already subtracting `STD_MAX_MATCH` which is a lot more. Those guard bytes should be in addition to the window size, not included in it.

Reply:
Great point. Let me look into it.

Reply:
Ok I have removed the `safe` variable altogether in my most recent commit. Good find.